### PR TITLE
First pass at Languages/REPL

### PR DESCRIPTION
**Pull request by coke**
For  #4270

### CODE REVIEW

File: doc/Language/REPL.rakudoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+=item 1
+
+=for code :lang<shell>
+    zef install Terminal::LineEditor
+
+=item 2
+
+=for code :lang<shell>
+    zef install Linenoise
```
Comment (patrickbkr) [inline doc/Language/REPL.rakudoc:56]: Mention that this requires a working C build toolchain.

File: doc/Language/REPL.rakudoc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+In a fresh install, there is no command history. Running the C<REPL> in this mode will
+prompt you to install one of the various modules that provide this support. If enabled,
+you can use the arrow keys to scroll through previous commands and use standard terminal
+shortcuts for editing. There are currently four options:
+
+=item 1
+
+=for code :lang<shell>
+    zef install Terminal::LineEditor
```
Comment (patrickbkr) [inline doc/Language/REPL.rakudoc:51]: Technically this also requires a working C build toolchain at the moment, but I wouldn't mention that. It's a temporary thing. It depends on `Term::termios`, which has C code in it, but none of the `Terminal::*` modules require the functionality of that module that uses the C code. I plan to change this (possibly by creating a `Term::termios::PureRaku` or so), if no one beats me to it.

Comment (coke) [reply doc/Language/REPL.rakudoc:51]: Internalized. :)

File: doc/Language/REPL.rakudoc (hunk 1 above)
Comment (patrickbkr) [reply doc/Language/REPL.rakudoc:56]: This is not right.

Readline does *not* require a C compiler, but *does* require the readline library to be installed.

Linenoise *does* require a C compiler as it contains and builds a full copy of linenoise (which is a C library).

Comment (coke) [reply doc/Language/REPL.rakudoc:56]: Resolved

**Pull request merged**